pkg/syncutil/cachemanager/cachemanager_test.go

```diff
@@ -0,0 +1,138 @@
+package cachemanager
```

Comment:
Let's remember that `config_controller_test`, by definition, tests the sync controller, which will test that the CM handles the current uses cases as well.

So the new tests here are meant to only target the `cachemanager` in as much isolation as possible from controllers.

Another point to remember is that as we build the background replay process more into the `cachemanager`, the tests will change to cover the new functionality.